Opensds_state_machine_design.md

```diff
@@ -0,0 +1,109 @@
+# Opensds state machine design
+
+Status: WIP
+
+Version: Alpha
+
+Author(s): jack li (@lasshpanzer)
+
+## Summary
+
+The State machines of volume,snapshot and attachment show the real-time status and state transitions between actions.
+
+## Motivation
+
+For example, When creating a volume, there is needed to know if it has been created or if something went wrong.
+
+### Goals
+
+Design volume statuses
+
+Design snapshot statuses
+
+Design attachment statuses
+
+### Non-Goals
+
+Some architectures needs to be changed.
+
+## Proposal
+
+There is no mechanism to show the real-time status and state transitions in the current opensds version.It needs to be implemented.
+
+### Data model impact
+
+The state attributes of the data have been added to many new states
+
+### REST API impact
+
+Does the proposal affect API?
+No
+
+### Security impact
+
+Does it affect security?
+No
+
+### Other end user impact
+
+Other than API, are there other changes that affect a user? For example the CLI changes.
```

Comment:
When getting a resource such as a volume, end user can now see the changed status of the resource.